python/lsst/drp/tasks/metadetection_shear.py

```diff
@@ -252,37 +578,405 @@ def run(self, patch_coadds: Sequence[MultipleCellCoadd]) -> Struct:
+        fake_noise_image.array[:, :] = self.rng.normal(
+            scale=noise,
+            size=fake_noise_image.array.shape,
+        )
+        return self._make_cell_exposure(
+            fake_noise_image, cell_coadd,
+        )
+        # return self._make_cell_exposure(
+        #     cell_coadd.outer.noise_realizations[index], cell_coadd,
+        # )
+
+    def _make_mfrac_exposure(self, cell_coadd: SingleCellCoadd) -> ExposureF:
+        # TODO: cell coadds will have a real mfrac image
+        # True means deep copy
+        fake_mfrac_image = ImageF(cell_coadd.outer.image, True)
+        fake_mfrac_image.array[:, :] = 0
+        return self._make_cell_exposure(
+            fake_mfrac_image, cell_coadd,
+        )
+
+    def _make_cell_exposure(
+        self, image: ImageF, cell_coadd: SingleCellCoadd,
+    ) -> ExposureF:
+
+        masked_image = MaskedImageF(
+            image, cell_coadd.outer.mask, cell_coadd.outer.variance,
+        )
+
+        exposure = ExposureF(masked_image)
+        exposure.setWcs(cell_coadd.common.wcs)
+
+        psf = self._get_cell_psf(cell_coadd)
+        exposure.setPsf(psf)
+
+        band = cell_coadd.common.band
+        filter_label = FilterLabel(band=band, physical=band)
+        exposure.setFilter(filter_label)
+
+        return exposure
+
+    def _get_cell_psf(self, cell_coadd: SingleCellCoadd) -> KernelPsf:
+
+        # this makes a deep copy
+        psf_image = cell_coadd.psf_image.convertD()
+        psf_array = psf_image.array
+
+        wbad = np.where(~np.isfinite(psf_array))
+        if wbad[0].size == psf_array.size:
+            raise ValueError('no good pixels in the psf')
+
+        if wbad[0].size > 0:
+            self.log.debug('zeroing %d bad psf pixels' % wbad[0].size)
+            psf_array[wbad] = 0.0
+
+        psf_array *= 1/psf_array.sum()
+        # assert np.all(psf_image.array[:, :] == psf_array)
+        kernel = FixedKernel(psf_image)
+        return KernelPsf(kernel)
+
+
+def _dictify(data):
+    output = {}
+    for name in data.dtype.names:
+        output[name] = data[name]
+    return output
+
+
+def _make_comb_data(
+    cell_coadd,
+    res,
+    meas_type,
+    mask_frac,
+    full_output=False,
+):
+    import esutil as eu
+
+    idinfo = cell_coadd.identifiers
+
+    copy_dt = [
+        # we will copy out of arrays to these
+        ('psfrec_g_1', 'f4'),
+        ('psfrec_g_2', 'f4'),
+        ('wmom_psf_g_1', 'f4'),
+        ('wmom_psf_g_2', 'f4'),
+        ('wmom_g_1', 'f4'),
+        ('wmom_g_2', 'f4'),
+        ('wmom_band_flux_flags_1', 'i4'),
+        ('wmom_band_flux_1', 'f4'),
+        ('wmom_band_flux_err_1', 'f4'),
+    ]
+
+    add_dt = [
+        ('id', 'u8'),
+        ('tract', 'u4'),
+        ('patch_x', 'u1'),
+        ('patch_y', 'u1'),
+        ('cell_x', 'u1'),
+        ('cell_y', 'u1'),
+        ('shear_type', 'U2'),
+        ('mask_frac', 'f4'),
+        ('primary', bool),
```

Comment:
I don't see these items in the schema above?

Reply:
Note also that the code that creates the final output on disk seems to ignore fields that are not in the schema.